core/src/main/java/io/questdb/griffin/engine/functions/date/ToUTCTimestampFunctionFactory.java

```diff
@@ -0,0 +1,125 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.date;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.TimestampFunction;
+import io.questdb.std.*;
+import io.questdb.std.datetime.microtime.TimestampFormatUtils;
+import io.questdb.std.datetime.microtime.Timestamps;
+import org.jetbrains.annotations.NotNull;
+
+import static io.questdb.std.datetime.TimeZoneRuleFactory.RESOLUTION_MICROS;
+
+public class ToUTCTimestampFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "to_utc(NS)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        Function timestamp = args.getQuick(0);
+        Function timezone = args.getQuick(1);
+
+        if (timezone.isConstant()) {
+            return getTimestampFunction(argPositions, timestamp, timezone, -1);
+        } else {
+            return new ToTimezoneFunctionVar(timestamp, timezone);
+        }
+    }
+
+    @NotNull
+    static TimestampFunction getTimestampFunction(IntList argPositions, Function timestamp, Function timezone, int multiplier) throws SqlException {
+        final CharSequence tz = timezone.getStr(null);
+        if (tz != null) {
+            final int hi = tz.length();
+            final long l = Timestamps.parseOffset(tz, 0, hi);
+            if (l == Long.MIN_VALUE) {
+                try {
+                    return new OffsetTimestampFunctionFromRules(
+                            timestamp,
+                            TimestampFormatUtils.enLocale.getZoneRules(
+                                    Numbers.decodeLowInt(TimestampFormatUtils.enLocale.matchZone(tz, 0, hi)), RESOLUTION_MICROS
+                            ),
+                            multiplier
+                    );
+                } catch (NumericException e) {
+                    Misc.free(timestamp);
+                    throw SqlException.$(argPositions.getQuick(1), "invalid timezone name");
+                }
+            } else {
+                return new OffsetTimestampFunctionFromOffset(
+                        timestamp,
+                        multiplier * Numbers.decodeLowInt(l) * Timestamps.MINUTE_MICROS
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).